src/core/util/sys_vars.cpp

```diff
@@ -2744,6 +2754,16 @@ void mce_sys_var::configure_memory_limits(const config_registry &registry)
         tcp_timer_resolution_msec = timer_resolution_msec;
     }
 
+    // RFC 1122 Section 4.2.3.2: Delayed ACK timer must not exceed 500ms
+    // This limits TCP timer resolution to ensure protocol compliance and proper RTO calculations
+    if (tcp_timer_resolution_msec > 500) {
+        vlog_printf(VLOG_WARNING,
+                    "TCP timer resolution [%s=%d] exceeds RFC 1122 maximum of 500ms. "
+                    "Clamping to 500ms to ensure protocol compliance.\n",
+                    SYS_VAR_TCP_TIMER_RESOLUTION_MSEC, tcp_timer_resolution_msec);
+        tcp_timer_resolution_msec = 500;
+    }
```

Comment:
If we decide to keep the upper limit, we don't need this check because it's in the schema.

[comment] We don't have upper limit for the basic timer, right? In this case if timer > tcp_timer, limiting tcp_timer back to 500ms will calculate ticks based on this value while actual ticks will still run with the basic timer granularity.